aptos-move/framework/aptos-stdlib/sources/data_structures/big_ordered_map.move

```diff
@@ -0,0 +1,1400 @@
+        // (i.e. we should be using `reserved_slot` instead).
+        move node_index;
+
+        // Now we can perform the split at the current level, as we know we are not at the root level.
+        assert!(!path_to_node.is_empty(), error::invalid_state(EINTERNAL_INVARIANT_BROKEN));
+
+        // Parent has a reference under max key to the current node, so existing index
+        // needs to be the right node.
+        // Since ordered_map::trim moves from the end (i.e. smaller keys stay),
+        // we are going to put the contents of the current node on the left side,
+        // and create a new right node.
+        // So if we had before (node_index, node), we will change that to end up having:
+        // (new_left_node_index, node trimmed off) and (node_index, new node with trimmed off children)
+        //
+        // So let's rename variables cleanly:
+        let right_node_reserved_slot = reserved_slot;
+        let left_node = node;
+
+
+        let is_leaf = left_node.is_leaf;
+        let left_children = &mut left_node.children;
+
+        let right_node_index = right_node_reserved_slot.reserved_to_index();
+        let left_next = &mut left_node.next;
+        let left_prev = &mut left_node.prev;
+
+        // compute the target size for the left node:
+        let max_degree = if (is_leaf) {
+            self.leaf_max_degree as u64
+        } else {
+            self.inner_max_degree as u64
+        };
+        let target_size = (max_degree + 1) / 2;
+
+        // Add child (which will exceed the size), and then trim off to create two sets of children of correct sizes.
+        left_children.add(key, child);
+        let right_node_children = left_children.trim(target_size);
+
+        assert!(left_children.length() <= max_degree, error::invalid_state(EINTERNAL_INVARIANT_BROKEN));
+        assert!(right_node_children.length() <= max_degree, error::invalid_state(EINTERNAL_INVARIANT_BROKEN));
+
+        let right_node = new_node_with_children(is_leaf, right_node_children);
+
+        let (left_node_slot, left_node_reserved_slot) = self.nodes.reserve_slot();
+        let left_node_index = left_node_slot.stored_to_index();
+
+        // right nodes next is the node that was next of the left (previous) node, and next of left node is the right node.
+        right_node.next = *left_next;
+        *left_next = right_node_index;
+
+        // right nodes previous is current left node
+        right_node.prev = left_node_index;
+        // Since the previuosly used index is going to the right node, `prev` pointer of the next node is correct,
+        // and we need to update next pointer of the previous node (if exists)
+        if (*left_prev != NULL_INDEX) {
+            self.nodes.borrow_mut(*left_prev).next = left_node_index;
+        };
+        // Otherwise, we were the smallest node on the level. if this is the leaf level, update the pointer.
+        if (right_node_index == self.min_leaf_index) {
```

Comment:
nit: having an "else" is more readable, and we should assert "right_node_index == self.min_leaf_index" in the else branch, if I read it correctly?